frontend/src/components/organisms/EventForm.tsx

```diff
@@ -14,60 +15,314 @@ import FormControlLabel from "@mui/material/FormControlLabel";
+        }
+      : {}
+  );
+  const back = () => {
+    router.replace("/events/view");
+    setIsLoading(false);
+  };
+
+  /*fetch userId for ownerId*/
+  const fetchUserDetails = async () => {
+    try {
+      const fetchUrl = `${url}/users/search/?email=${user?.email}`;
+      const userToken = await auth.currentUser?.getIdToken();
+      const response = await fetch(fetchUrl, {
+        method: "GET",
+        headers: {
+          Authorization: `Bearer ${userToken}`,
+        },
+      });
+      if (response.ok) {
+        const data = await response.json();
+        return data["data"][0]["id"];
+      } else {
+        console.error("User Retrieval failed with status:", response.status);
+        setIsLoading(false);
+      }
+    } catch (error) {
+      console.log("Error in User Info Retrieval.");
+      console.log(error);
+    }
+  };
 
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+
+  const handleErrors = (errors: any) => {};
+
+  const CreateErrorComponent = (): JSX.Element | null => {
+    return errorMessage ? (
+      <Alert severity="error">Error: {errorMessage}</Alert>
+    ) : null;
+  };
+
+  const CreateSuccessComponent = (): JSX.Element | null => {
+    return successMessage ? (
+      <Alert severity="success">Success: {successMessage}</Alert>
+    ) : null;
+  };
+  /**Helper for handling creating events */
   const handleCreateEvent: SubmitHandler<FormValues> = async (data) => {
-    const {
-      eventName,
-      startDate,
-      endDate,
-      startTime,
-      endTime,
-      location,
-      volunteerSignUpCap,
-      eventDescription,
-      eventImage,
-      rsvpLinkImage,
-    } = data;
-    console.log(data);
+    setIsLoading(true);
+    const mode = status === 0 ? "VIRTUAL" : "IN_PERSON";
+    const startDateTime = convertToISO(getStartTime, getStartDate);
+    const endDateTime = convertToISO(getEndTime, getEndDate);
+    console.log("start time", getStartTime);
+    if (getStartTime == "") {
+      setErrorMessage("Start Time is required.");
+    } else if (getEndTime == "") {
+      setErrorMessage("End Time is required.");
+    } else if (getStartDate == "") {
+      setErrorMessage("Start Date is required.");
+    } else if (getEndDate == "") {
+      setErrorMessage("End Date is required.");
+    } else {
+      var dateValidation =
+        new Date(startDateTime) > new Date(endDateTime) ||
+        new Date(endDateTime) < new Date(startDateTime);
+      if (dateValidation) {
+        setErrorMessage("End Date cannot be earlier than Start Date");
+      } else {
+        setErrorMessage(null);
+      }
+    }
+    const { eventName, location, volunteerSignUpCap, eventDescription } = data;
+
+    const userid = await fetchUserDetails();
+    const fetchCreateEventsUrl = `${url}/events`;
+    try {
+      if (errorMessage) {
+        setIsLoading(false);
+        return;
+      }
```

Comment:
I think this should not be in the try catch. This causing errors like this where some form input is missing but it is showing use that it not able to create the event which means it send the request to  the backend anyway
<img width="981" alt="Screenshot 2023-11-20 at 16 37 39" src="https://github.com/cornellh4i/lagos-volunteers/assets/59776300/e87c5fb8-e26f-40e0-9e2a-cca7b0fb096f">